### PR TITLE
Add a custom separator input option to ini_manage.

### PR DESCRIPTION
**Pull request by boltronics**
Addresses issue #31209. Completely untested at this point, but it's a start if someone wants to take it from here.

### CODE REVIEW

**Comment by cachedout** [comment]
@boltronics This won't merge cleanly. Would you mind rebasing it?

File: salt/modules/ini_manage.py [hunk 1]
```diff
@@ -50,6 +50,10 @@ def set_option(file_name, sections=None):
         If the Ini does not contain sections the keys and values represent the
         options
 
+    separator : =
+        A character used to separate keys and values. Standard ini files use
+        the "=" character.
+
```
Comment (jfindlay) [inline salt/modules/ini_manage.py]: @boltronics, the documentation on this option should also include a `versionadded`:
```rst
    separator : =
        A character used to separate keys and values. Standard ini files use
        the "=" character.

        .. versionadded:: Carbon
```

Comment (boltronics) [inline salt/modules/ini_manage.py]: @cachedout Done, but I figure it should be properly tested before being merged in. I'll need to build a new salt and test state first, so will try to do that today or tomorrow.

@jfindlay, thanks for the versionadded comment hint. I've added such a line to the commit.

**Comment by boltronics** [comment]
I got around to setting up a test environment, and the execution module changes all seem to work for me with separator=' ' except for ini.remove_option which I need to investigate further.

**Comment by cachedout** [comment]
@boltronics Sounds good. Please let me know when you think this is all ready to go.

**Comment by boltronics** [comment]
@cachedout I think this is all good to go now. I've given all changes enough testing that I'm reasonably comfortable it's all working as intended.

Also notice a second commit I snuck in there. It looks like ini.options_absent never worked correctly, or at least not as documented (which would produce a "TypeError: unhashable type: 'list'" error). Removing the '-' from each line gets it working as intended, although this feels somewhat inconsistent.

**Comment by cachedout** [comment]
@boltronics Looks like some spurious pylint errors snuck in there. Could you add some pylint disable lines to silence those? The rest looks good. Once that's done, I'll get this in. Thanks!

**Comment by boltronics** [comment]
@cachedout It looks like switching to the new-style Python string substitutions sorted this out without needing to add the ```# pylint: disable=E1321``` lines.

**Comment by rallytime** [comment]
Thanks @boltronics!

**Pull request merged**